mqtt-cs/src/main/java/org/apache/rocketmq/mqtt/cs/channel/DefaultChannelManager.java

```diff
@@ -136,6 +141,10 @@ public void closeConnect(Channel channel, ChannelCloseFrom from, String reason)
         if (channel.isActive()) {
             channel.close();
         }
+
+        // add client offline event
+        clientEventHookManager.putClientEvent(channel, ClientEventType.DISCONNECT);
```

Comment:
Considering adding the reason of disconnection into disconnect event?

Reply:
Thx. I will add `eventInfo` to satisfy this case.